doc/source/ray-job-submission/overview.rst

```diff
@@ -0,0 +1,239 @@
+        }
+    )
+
+.. tip::
+
+    By default ray job server will generate a new uuid as return value, but you can also generate your unique job_id first and pass it into :code:`submit_job`. In this case the job will be executed with your given id, and will throw error if same job_id is submitted more than once for the same ray cluster. This can facilitate job id management assuming you have your own job registry to generate and persist unique job ids.
+
+Now we can have a simple polling loop that checks job status until it reaches terminal state, and get logs at the end. We are expected to see actor printed numbers as well as correct version of :code:`requests` module used via runtime_env.
+
+.. code-block:: python
+
+    from ray.dashboard.modules.job.common import JobStatus, JobStatusInfo
+
+    def wait_until_finish(job_id):
+        start = time.time()
+        timeout = 5
+        while time.time() - start <= timeout:
+            status_info = client.get_job_status(job_id)
+            status = status_info.status
+            print(f"status: {status}")
+            if status in {JobStatus.SUCCEEDED, JobStatus.STOPPED, JobStatus.FAILED}:
+                break
+            time.sleep(1)
+
+
+    wait_until_finish(job_id)
+    logs = client.get_job_logs(job_id)
+    print(logs)
+
+.. tip::
+
+    We can also use other remote uris for runtime env, such as S3 or GIT. See "Remote URIs" section of :ref:`Runtime Environments<runtime-environments>` for details.
+
+Submitted job can be stopped by user before finish executing.
+
+.. code-block:: python
+
+    job_id = client.submit_job(
+        # Entry point to execute
+        entrypoint="python -c 'import time; time.sleep(60)'",
+        runtime_env={}
+    )
+    wait_until_finish(job_id)
+    client.stop_job(job_id)
+    wait_until_finish(job_id)
+    logs = client.get_job_logs(job_id)
+    print(logs)
+
+
+Job CLI API
+-----------
+
+In addition to job SDK, we can also submit ray application via CLI.
+
+
+| Ensure we have a local ray cluster with running headnode.
+
+.. code-block:: bash
+
+   ray start --head
+
+.. code-block:: python
+
+    ❯ ray job submit --address="127.0.0.1:8265" -- "python -c 'print(123); import time; time.sleep(5)'"
+    2021-11-18 16:14:47,602	INFO cli.py:103 -- Job submitted successfully: raysubmit_GsQYzyvZpgNicU8F.
+    2021-11-18 16:14:47,602	INFO cli.py:104 -- Query the status of the job using: `ray job status raysubmit_GsQYzyvZpgNicU8F`.
+
+
+    ❯ ray job status raysubmit_GsQYzyvZpgNicU8F
+    2021-11-18 16:15:07,727	INFO cli.py:125 -- Job status for 'raysubmit_GsQYzyvZpgNicU8F': SUCCEEDED.
+    2021-11-18 16:15:07,727	INFO cli.py:127 -- Job finished successfully.
+
+
+    ❯ ray job logs raysubmit_GsQYzyvZpgNicU8F
+    123
+
+
+Job HTTP API
+------------
+
+Under the hood, both Job Client and CLI make HTTP calls to the job server running on ray head node. Therefore user can also directly send requests to corresponding endpoints via HTTP if needed.
```

Comment:
```suggestion
Under the hood, both the Job Client and the CLI make HTTP calls to the job server running on the ray head node. Therefore the user can also directly send requests to corresponding endpoints via HTTP if needed.
```

Maybe say Job SDK here instead of Job Client to reduce the number of different terms?